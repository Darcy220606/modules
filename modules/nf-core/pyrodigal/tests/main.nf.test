nextflow_process {

    name "Test Process PYRODIGAL"
    script "../main.nf"
    process "PYRODIGAL"

    tag "modules"
    tag "modules_nfcore"
    tag "pyrodigal"

    test("pyrodigal - sarscov2 - gbk") {
        when {
            process {
                """
                input[0] =  [
                             [id:'test', single_end:false ], // meta map
                             file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                 input[1] =  'gbk'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("pyrodigal - sarscov2 - gbk - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] =  [
                             [id:'test', single_end:false ], // meta map
                             file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                            ]
                input[1] =  'gbk'
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.annotations.collect { file(it[1]).getName() } +
                                    process.out.fna.collect { file(it[1]).getName() } +
                                    process.out.faa.collect { file(it[1]).getName() } +
                                    process.out.score.collect { file(it[1]).getName() } +
                                    process.out.versions).match() }
            )
        }
    }

}