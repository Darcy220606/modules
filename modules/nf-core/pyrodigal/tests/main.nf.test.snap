{
    "pyrodigal - sarscov2 - gff - stub": {
        "content": [
            [
                "test.gff.gz",
                "test.fna.gz",
                "test.faa.gz",
                "test.score.gz",
                "versions.yml:md5,4aab54554829148e01cc0dc7bf6cb5d3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T23:14:56.524595168"
    },
    "pyrodigal - sarscov2 - gff": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,8f8b3c876d6691e09828f883da32ae26"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fna.gz:md5,56567e79600268bb968b48ca6b80c2aa"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,3e3e40ccc26ba4d5dd5c6057782906c7"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.score.gz:md5,f38518f49694b7a73883fa4b513a522f"
                    ]
                ],
                "4": [
                    "versions.yml:md5,4aab54554829148e01cc0dc7bf6cb5d3"
                ],
                "annotations": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.gff.gz:md5,8f8b3c876d6691e09828f883da32ae26"
                    ]
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.faa.gz:md5,3e3e40ccc26ba4d5dd5c6057782906c7"
                    ]
                ],
                "fna": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fna.gz:md5,56567e79600268bb968b48ca6b80c2aa"
                    ]
                ],
                "score": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.score.gz:md5,f38518f49694b7a73883fa4b513a522f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4aab54554829148e01cc0dc7bf6cb5d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-17T23:14:48.445244303"
    }
}